commands/config.js

```diff
@@ -21,63 +21,90 @@ exports.run = async (client, message, args, settings) => {
             break;
         }
         case 'welcomeChannel': {
-            /**
-             * Channel validation. Check if the mentioned channel is a guild.
-             * Want a hint? 
-             * ```js
-             * let channel = message.guild.channels.find(c => c.name === updated);
-             * 
-             * // example of Collection#find (look below)
-             * collection.find(val => val.username === 'Anthony');
-             * ```
-             * Remember when I talked about collections earlier in the video!
-             * https://discord.js.org/#/docs/main/stable/class/Collection?scrollTo=find
-             * https://anidiots.guide/understanding/collections
-             */
+             if (updated) {
+               try {
+                const channel = message.guild.channels.find(c => c.name === updated) || message.mentions.channels.first();
+                if (!channel) return message.reply('could not find channel');
 
+                await client.updateGuild(message.guild, { welcomeChannel: channel.id });
+                return message.channel.send(`Updated channel to: ${channel}`);
+               } catch (e) {
+                   console.log(e);
+                   message.channel.send(`An error occurred: **${e.message}**`);
+               }
+             }
+                message.reply(`Usage: ${client.config.prefix}config welcomeChannel #<Channel> or the name of the channel`);
+             
             break;
         }
         case 'welcomeMsg': {
-            /**
-             * Make sure the user specifically defines the {{user}} and {{guild}} parameters.
-             * Want a hint?
-             * ```js
-             * let foo = '{{bar}}';
-             * let message = 'Hello, {bar}';
-             * 
-             * if (foo.test(message)) {
-             *  console.log('Wooo');
-             * } else {
-             *  console.log('No...');
-             * }
-             * ```
-             */
+            if (updated) {
+                try {
+                    if (!updated.includes('{{user}}' && '{{guild}}')) {
```

Comment:
For validating the user input, you should be using regular expressions (regex) and not `String.prototype.includes()`. The reason being is by using Regex, you have finer control than `.includes()`. For example, if a user inputted this string below as the welcome message, `.includes()` would still return `true`:

Hello {{user}}**}**! Welcome to {{guild}}!

Do you see where I added that extra `}`? With a proper Regex, you could make sure the user isn't adding anything extra that they shouldn't be. It's good for validation! If you're curious about reading more, I would check out this article (even helped me get a better understanding): https://love2dev.com/blog/javascript-includes/

Reply:
Oh okay Thanks for the great reply :)